pkg/networks/commands.go

```diff
@@ -0,0 +1,84 @@
+package networks
+
+import (
+	"fmt"
+	"github.com/lima-vm/lima/pkg/store"
+)
+
+const (
+	Switch = "switch"
+	VMNet  = "vmnet"
+)
+
+// Commands in `sudoers` cannot use quotes, so all arguments are printed via "%s"
+// and not "%q". config.Paths.* entries must not include any whitespace!
+
+func (config *NetworksConfig) Check(name string) error {
+	if _, ok := config.Networks[name]; ok {
+		return nil
+	}
+	return fmt.Errorf("network %q is not defined", name)
+}
+
+func (config *NetworksConfig) VDESock(name string) string {
+	return fmt.Sprintf("%s/%s.ctl", config.Paths.VarRun, name)
+}
+
+func (config *NetworksConfig) PIDFile(name, daemon string) string {
+	return fmt.Sprintf("%s/%s_%s.pid", config.Paths.VarRun, name, daemon)
+}
+
+func (config *NetworksConfig) LogFile(name, daemon, stream string) string {
+	networksDir, _ := store.LimaNetworksDir()
+	return fmt.Sprintf("%s/%s_%s.%s.log", networksDir, name, daemon, stream)
+}
+
+func (config *NetworksConfig) DaemonUser(daemon string) string {
+	switch daemon {
+	case Switch:
+		return "daemon"
+	case VMNet:
+		return "root"
+	}
+	panic("daemonuser")
+}
+
+func (config *NetworksConfig) DaemonGroup(daemon string) string {
+	switch daemon {
+	case Switch:
+		return config.Group
+	case VMNet:
+		return "wheel"
+	}
+	panic("daemongroup")
+}
+
+func (config *NetworksConfig) MkdirCmd() string {
+	return fmt.Sprintf("/bin/mkdir -m 775 -p %s", config.Paths.VarRun)
+}
+
+func (config *NetworksConfig) StartCmd(name, daemon string) string {
+	var cmd string
+	switch daemon {
+	case Switch:
+		cmd = fmt.Sprintf("%s --pidfile=%s --sock=%s --group=%s --dirmode=0770 --nostdin",
+			config.Paths.VDESwitch, config.PIDFile(name, Switch), config.VDESock(name), config.Group)
+	case VMNet:
+		nw := config.Networks[name]
+		cmd = fmt.Sprintf("%s --pidfile=%s --vde-group=%s --vmnet-mode=%s",
```

Comment:
Can we error out when the `vde_vmnet` binary (or its DLLs) is modifiable by non-root?

Reply:
Yes, I intended to do this as part of `some validation for networks.yaml`. Not sure about the dylibs though; I don't want to shell out to `otool` to look at the import paths, and then figure out how to resolve `@executable_path/...` references etc. Let me know how you would want to determine the library dependencies!

I was planning to make sure all `config.Path` entries don't have any whitespace, and the files and all parent etc. directories are only writable by root.

Reply:
The otool stuff can be another PR in future.

Reply:
> Can we error out when the `vde_vmnet` binary (or its DLLs) is modifiable by non-root?

Done in b6bcc80ba862b348839f62842bee9dd019550dec

(otool logic to come in a later PR)

Reply:
> (otool logic to come in a later PR)

I found that `/usr/bin/otool` is just a shim that tries to locate `XCode.app`, and if it can't find it presents a dialog offering to install the "commandline developer tools".

So it is going to be tricky to figure out if it is even safe to call `otool`. And what do we do when it is not available? Fall back on checking the "known filenames"?

Reply:
Maybe we can parse macho directly (in another PR)

Reply:
This should work https://pkg.go.dev/debug/macho#File.ImportedLibraries